main.nf

```diff
@@ -391,8 +395,10 @@ if( params.aligner == 'bismark' ){
         file wherearemyfiles from ch_wherearemyfiles_for_bismark_align.collect()
 
         output:
-        set val(name), file("*.bam") into ch_bam_for_bismark_deduplicate, ch_bam_for_bismark_summary, ch_bam_for_preseq
-        set val(name), file("*report.txt") into ch_bismark_align_log_for_bismark_report, ch_bismark_align_log_for_bismark_summary, ch_bismark_align_log_for_multiqc
+        file("*.bam") into ch_bam_for_merge
```

Comment:
I don't think you need any of these changes. Just switching `ch_bam_for_bismark_deduplicate` for `ch_bam_for_merge` should be sufficient.